components/tools/OmeroJava/test/integration/LightAdminRolesTest.java

```diff
@@ -2437,4 +2447,160 @@ public void testModifyGroupEdit(boolean isPrivileged,
+     * This should fail.
+     * @throws Exception
+     */
+    @Test(expectedExceptions = omero.ValidationException.class)
+    public void testDeleteUsedEnumerationsbyRestrictedSystemUser() throws Exception {
+        final IAdminPrx svc = root.getSession().getAdminService();
+        final List<AdminPrivilege> expectedPrivileges = new ArrayList<AdminPrivilege>();
+        final String lightAdminName = UUID.randomUUID().toString();
+        final String lightAdminPassword = UUID.randomUUID().toString();
+        Experimenter lightAdmin = createExperimenterI(lightAdminName, "test", "user");
+        final long lightAdminId = svc.createRestrictedSystemUserWithPassword(lightAdmin,
+                expectedPrivileges, omero.rtypes.rstring(lightAdminPassword));
+        lightAdmin = svc.getExperimenter(lightAdminId);
+        omero.client client = newOmeroClient();
+        client.createSession(lightAdminName, lightAdminPassword);
+        init(client);
+        //try to delete an enum type. The type is not important
+        final ITypesPrx types_svc = factory.getTypesService();
+        List<IObject> types = types_svc.allEnumerations(EventType.class.getName());
+        Iterator<IObject> i = types.iterator();
+        while (i.hasNext()) {
+            types_svc.deleteEnumeration(i.next());
+        }
+    }
+
+    /**
+     * Tests if a deleted enumeration can be restored.
+     * @throws Exception
+     */
+    @Test
+    public void testResetEnumerationsbyRestrictedSystemUser() throws Exception {
+        final IAdminPrx svc = root.getSession().getAdminService();
+        final List<AdminPrivilege> expectedPrivileges = new ArrayList<AdminPrivilege>();
+        final String lightAdminName = UUID.randomUUID().toString();
+        final String lightAdminPassword = UUID.randomUUID().toString();
+        Experimenter lightAdmin = createExperimenterI(lightAdminName, "test", "user");
+        final long lightAdminId = svc.createRestrictedSystemUserWithPassword(lightAdmin,
+                expectedPrivileges, omero.rtypes.rstring(lightAdminPassword));
+        lightAdmin = svc.getExperimenter(lightAdminId);
+        omero.client client = newOmeroClient();
+        client.createSession(lightAdminName, lightAdminPassword);
+        init(client);
+        final ITypesPrx types_svc = factory.getTypesService();
+        List<IObject> types = types_svc.allEnumerations(ContrastMethod.class.getName());
+        int n = types.size();
+        Iterator<IObject> i = types.iterator();
+        while (i.hasNext()) {
+            types_svc.deleteEnumeration(i.next());
+        }
+        types = types_svc.allEnumerations(ContrastMethod.class.getName());
+        Assert.assertEquals(types.size(), 0);
+        types_svc.resetEnumerations(ContrastMethod.class.getName());
+        types = types_svc.allEnumerations(ContrastMethod.class.getName());
+        Assert.assertEquals(types.size(), n);
+        types_svc.resetEnumerations(ContrastMethod.class.getName());
```

Comment:
This reset might also be good for `root` to do in an `@AfterClass` depending on in what state these test methods may have left the server.

Reply:
True but here I wanted to check that any member of system could roll back potential mistakes and not only root